app/models/setting.rb

```diff
@@ -232,8 +232,13 @@ def self.model_name
   def validate_host_owner
     owner_type_and_id = value
     return if owner_type_and_id.blank?
-    owner = OwnerClassifier.new(owner_type_and_id).user_or_usergroup
-    errors.add(:value, _("Host owner is invalid")) if owner.nil?
+
+    begin
+      owner = OwnerClassifier.new(owner_type_and_id).user_or_usergroup
+      raise StandardError, _("Host owner is invalid. Please ensure the input follows the format '${id}-[Users|Usergroups]'.") if owner.nil?
+    rescue StandardError => e
+      errors.add(:value, e.message)
```

Comment:
Why would you raise an exception and immediately rescue it?
You can add a more detailed message to the `errors` array instead.

Reply:
I suggested that `OwnerClassifier` would raise the exception, not the model.